src/rebalancers/libraries/RebalanceLogic.sol

```diff
@@ -0,0 +1,252 @@
+     *   - If the current price is below the position, similar, we swap the full position to token0.
+     *   - If the position is in range we proceed with step 2.
+     *
+     * 2) If the position is in range, we start with calculating the "Target Ratio" and "Current Ratio".
+     *    Both ratio's are defined as the value of the amount of token1 compared to the total value of the position:
+     *    R = valueToken1 / [valueToken0 + valueToken1]
+     *    If we express all values in token1 and use the current pool price to denominate token0 in token1:
+     *    R = amount1 / [amount0 * sqrtPrice² + amount1]
+     *
+     *    a) The "Target Ratio" (R_target) is the ratio of the new liquidity position.
+     *       It is calculated with the current price and the upper and lower prices of the liquidity position,
+     *       see _getTargetRatio() for the derivation.
+     *       To maximise the liquidity of the new position, the balances after the swap should approximate it as close as possible to not have any leftovers.
+     *    b) The "Current Ratio" (R_current) is the ratio of the current token balances, it is calculated as follows:
+     *       R_current = balance1 / [balance0 * sqrtPrice² + balance1].
+     *
+     * 3) From R_target and R_current we can finally derive the direction of the swap, amountIn and amountOut.
+     *    If R_current is smaller than R_target, we have to swap an amount of token0 to token1, and vice versa.
+     *    amountIn and amountOut can be found by solving the following equalities:
+     *      a) The ratio of the token balances after the swap equal the "Target Ratio".
+     *      b) The swap between token0 and token1 is done in the pool itself,
+     *         taking into account fees, but ignoring slippage (-> sqrtPrice remains constant).
+     *
+     *    - If R_current < R_target (swap token0 to token1):
+     *      a) R_target = [amount1 + amoutOut] / [(amount0 - amountIn) * sqrtPrice² + (amount1 + amoutOut)].
+     *      b) amountOut = (1 - fee) * amountIn * sqrtPrice².
+     *         => amountOut = [(R_target - R_current) * (amount0 * sqrtPrice² + amount1)] / [1 + R_target * fee / (1 - fee)].
+     *
+     *    - If R_current > R_target (swap token1 to token0):
+     *      a) R_target = [(amount1 - amountIn)] / [(amount0 + amoutOut) * sqrtPrice² + (amount1 - amountIn)].
+     *      b) amountOut = (1 - fee) * amountIn / sqrtPrice².
+     *         => amountIn = [(R_current - R_target) * (amount0 * sqrtPrice² + amount1)] / (1 - R_target * fee).
+     */
+    function _getSwapParams(
+        uint256 sqrtPrice,
+        uint256 sqrtRatioLower,
+        uint256 sqrtRatioUpper,
+        uint256 balance0,
+        uint256 balance1,
+        uint256 fee
+    ) internal pure returns (bool zeroToOne, uint256 amountIn, uint256 amountOut) {
+        if (sqrtPrice >= sqrtRatioUpper) {
+            // Position is out of range and fully in token 1.
```

Comment:
Would adapt comment here and below to make more clear it's related to the NEW position ticks (otherwise can be confusing). _New position is a single-sided liquidity position in token 1_.